Jellyfin.Networking/HappyEyeballs/HttpClientExtension.cs

```diff
@@ -72,45 +28,46 @@ public static bool UseIPv6
         /// <returns>The http steam.</returns>
         public static async ValueTask<Stream> OnConnect(SocketsHttpConnectionContext context, CancellationToken cancellationToken)
         {
-            // Until .NET supports an implementation of Happy Eyeballs (https://tools.ietf.org/html/rfc8305#section-2),
-            // let's make IPv4 fallback work in a simple way. This issue is being tracked at https://github.com/dotnet/runtime/issues/26177
-            // and expected to be fixed in .NET 6.
-            if (UseIPv6)
+            if (!UseIPv6)
             {
-                try
-                {
-                    var localToken = cancellationToken;
+                return await AttemptConnection(AddressFamily.InterNetwork, context, cancellationToken).ConfigureAwait(false);
+            }
 
-                    if (!_hasResolvedIPv6Availability)
-                    {
-                        // to make things move fast, use a very low timeout for the initial ipv6 attempt.
-                        using var quickFailCts = new CancellationTokenSource(ConnectionEstablishTimeout);
-                        using var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, quickFailCts.Token);
+            using var cancelIPv6 = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+            var tryConnectAsyncIPv6 = AttemptConnection(AddressFamily.InterNetworkV6, context, cancelIPv6.Token);
 
-                        localToken = linkedTokenSource.Token;
-                    }
+            if (await Task.WhenAny(tryConnectAsyncIPv6, Task.Delay(200, cancelIPv6.Token)).ConfigureAwait(false) == tryConnectAsyncIPv6 && tryConnectAsyncIPv6.IsCompletedSuccessfully)
+            {
+                cancelIPv6.Cancel();
+                return tryConnectAsyncIPv6.GetAwaiter().GetResult();
```

Comment:
Why GetAwaiter.GetResult instead of await?

Reply:
The task has already completed so it won't have any performance effect. It also apparently is better at handling exceptions according to [this comment](https://github.com/dotnet/corefx/pull/29792/files#r189415885).

Reply:
Sorry for letting it sit for so long. Can you add a code comment explaining this?